### ISSUE TITLE
Cannot configure project using CMakeTools on previously worked project

### ISSUE THREAD

**Issue by noamotto** [vector-of-bool/vscode-cmake-tools#183]
I tried to build a project I'm working on using CMakeTools and got the following strange error message:

> CMakeTools extension was unable to initialize: AssertionError: The specified configuration type is missing. The tools for the [See output window for more details]

When looking in the output window I found only this log message:

> Loading CMake Tools from source/path/.vscode/.cmaketools.json

I used to build this project before with previous versions of CMakeTools, so I believe it has something to do with one of the newest updates.

EDIT:

I looked into the debug message, and got 2 errors on startup:
```
shell.ts:445 Invalid backend promise: Error: Invalid backend promise
    at new CMakeToolsWrapper (C:\Users\Owner\.vscode\extensions\vector-of-bool.cmake-tools-0.10.0\out\src\wrapper.js:46:40)
    at C:\Users\Owner\.vscode\extensions\vector-of-bool.cmake-tools-0.10.0\out\src\extension.js:17:23
    at Generator.next (<anonymous>)
    at C:\Users\Owner\.vscode\extensions\vector-of-bool.cmake-tools-0.10.0\out\src\extension.js:7:65
    at __awaiter (C:\Users\Owner\.vscode\extensions\vector-of-bool.cmake-tools-0.10.0\out\src\extension.js:3:12)
    at activate (C:\Users\Owner\.vscode\extensions\vector-of-bool.cmake-tools-0.10.0\out\src\extension.js:15:12)
    at Function.t._callActivateOptional (d:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:23:418399)
    at Function.t._callActivate (d:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:23:418184)
    at d:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:23:417959
    at Object.m [as _notify] (d:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:23:39252)
```

And right after that:
```
shell.ts:445 The specified configuration type is missing.  The tools for the: AssertionError: The specified configuration type is missing.  The tools for the
    at Console.assert (console.js:95:23)
    at output.split.map.filter.reduce (C:\Users\Owner\.vscode\extensions\vector-of-bool.cmake-tools-0.10.0\out\src\environment.js:74:21)
    at Array.reduce (native)
    at C:\Users\Owner\.vscode\extensions\vector-of-bool.cmake-tools-0.10.0\out\src\environment.js:72:14
    at Generator.next (<anonymous>)
    at fulfilled (C:\Users\Owner\.vscode\extensions\vector-of-bool.cmake-tools-0.10.0\out\src\environment.js:4:58)
```

**Comment by ytimenkov** [comment]
This is a duplicate of #181 and others: error message from `vcvarsall.bat` is handled incorrectly by CMake Tools.

Full message is "The specified configuration type is missing.  The tools for the configuration might not be installed.". It is printed when corresponding vcvars<arch>.bat is missing (likely a partial installation).

If you have VS 2017 please try running VS Code with removed `VS140COMNTOOLS` environment variables as a temporary workaround.

**Comment by noamotto** [comment]
I used the workaround, and managed to configure my projects and use the extension. But when I tried to debug one of my tests. Then I got the message:

> Unable to start debugging. Launch options string provided by the project system is invalid. Unable to determine path to debugger. Please specify the "MIDebuggerPath" option.

Now I googled this variable and found that it supposed to use this variable to set the path to the debugger. As far as I know, the extension is supposed to set this variable (correct me if I'm wrong). Is it unset due to the workaround? What should I set there?
